### PR TITLE
modules/vpc: support re-apply of terraform when aws AZ number changes

### PR DESCRIPTION
**Pull request by trawler**

### CODE REVIEW

**Comment by trawler** [comment]
ok to test

**Comment by trawler** [comment]
aws green

**Comment by trawler** [comment]
retest this please. govcloud.

File: modules/aws/vpc/common.tf [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  new_master_az_count = "${ local.external_vpc_mode ? 0 : length(local.new_master_subnet_azs) }"
+
+  // The base set of ids needs to build rest of vpc data sources
+  // This is crux of dealing with existing vpc / new vpc incongruity
+  vpc_id = "${local.external_vpc_mode ? var.external_vpc_id : element(concat(aws_vpc.new_vpc.*.id,list("padding")),0)}"
+
+  // When referencing the _ids arrays or data source arrays via count = , always use the *_count variable rather than taking the length of the list
+  worker_subnet_ids   = ["${coalescelist(aws_subnet.worker_subnet.*.id,var.external_worker_subnet_ids)}"]
+  master_subnet_ids   = ["${coalescelist(aws_subnet.master_subnet.*.id,var.external_master_subnet_ids)}"]
+  worker_subnet_count = "${ local.external_vpc_mode ? length(var.external_worker_subnet_ids) : local.new_worker_az_count }"
```
Comment (squat) [inline modules/aws/vpc/common.tf]: I am a little surprised this is not accounted for in terraform fmt. can we remove these extra whitespaces?

File: modules/aws/vpc/common.tf [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  master_subnet_count = "${ local.external_vpc_mode ? length(var.external_master_subnet_ids) : local.new_master_az_count }"
```
Comment (squat) [inline modules/aws/vpc/common.tf]: and here as well

File: modules/aws/vpc/common.tf [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  new_worker_subnet_azs = ["${coalescelist(keys(var.new_worker_subnet_configs), data.aws_availability_zones.azs.names)}"]
+  new_master_subnet_azs = ["${coalescelist(keys(var.new_master_subnet_configs), data.aws_availability_zones.azs.names)}"]
+
+  // How many AZs to create worker and master subnets in (always zero if external_vpc_mode)
+  new_worker_az_count = "${ local.external_vpc_mode ? 0 : length(local.new_worker_subnet_azs) }"
+  new_master_az_count = "${ local.external_vpc_mode ? 0 : length(local.new_master_subnet_azs) }"
+
+  // The base set of ids needs to build rest of vpc data sources
+  // This is crux of dealing with existing vpc / new vpc incongruity
+  vpc_id = "${local.external_vpc_mode ? var.external_vpc_id : element(concat(aws_vpc.new_vpc.*.id,list("padding")),0)}"
```
[the remark this answers is not in the document]
Comment (trawler) [reply modules/aws/vpc/common.tf]: fixed

**Pull request merged**